### PR TITLE
Fix 404s on External Links

### LINKED ISSUE #8339: 404 external links
# Bug report

## Description

404 links

```
- _site/cloud/cdn/fastly-vcl-badreferer.html
  *  External link https://docs.fastly.com/guides/vcl-snippets/using-regular-vcl-snippets failed: 404 No error
- _site/guides/v2.3/coding-standards/docblock-standard-general.html
  *  External link https://docs.phpdoc.org/latest/references/phpdoc/basic-syntax.html#what-is-a-docblock failed: 404 No error
- _site/guides/v2.3/config-guide/secy/secy-xframe.html
  *  External link http://blogs.msdn.com/b/ieinternals/archive/2010/03/30/combating-clickjacking-with-x-frame-options.aspx failed: 404 No error
- _site/guides/v2.3/config-guide/varnish/config-varnish-advanced.html
  *  External link https://varnish-cache.org/docs/4.1/installation/install.html failed: 404 No error
  *  External link https://varnish-cache.org/docs/4.1/users-guide/vcl-backends.html?highlight=health%20check#health-checks failed: 404 No error
  *  External link https://www.varnish-cache.org/docs/4.1/reference/index.html failed: 404 No error
- _site/guides/v2.3/config-guide/varnish/config-varnish-install.html
  *  External link http://wiki.mikejung.biz/Varnish failed: got a time out (response code 301)
- _site/guides/v2.3/extension-dev-guide/cli-cmds/cli-howto.html
  *  External link https://symfony.com/doc/master/console/coloring.html failed: 404 No error
- _site/guides/v2.4/test/integration/annotations.html
  *  External link https://community.magento.com/t5/Magento-DevBlog/Indexers-parallelization-and-optimization/ba-p/104922_ failed: 400 No error
- _site/mftf/v2/docs/test/annotations.html
  *  External link https://devhub.io/zh/repos/allure-framework-allure-phpunit#extended-test-class-or-test-method-description failed: 404 No error
  *  External link https://devhub.io/zh/repos/allure-framework-allure-phpunit#human-readable-test-class-or-test-method-title failed: 404 No error
  *  External link https://devhub.io/zh/repos/allure-framework-allure-phpunit#map-test-classes-and-test-methods-to-features-and-stories failed: 404 No error
  *  External link https://devhub.io/zh/repos/allure-framework-allure-phpunit#set-test-severity failed: 404 No error
```

## Expected result

The 404 links are replaced with working links or deleted.

### PR DESCRIPTION
**Pull request by BarnyShergold**
## Purpose of this pull request

This pull request (PR) fixes #8339 

## Affected DevDocs pages
https://devdocs.magento.com/cloud/cdn/fastly-vcl-badreferer.html
https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html
https://devdocs.magento.com/guides/v2.3/config-guide/secy/secy-xframe.html
https://devdocs.magento.com/guides/v2.3/config-guide/varnish/config-varnish-advanced.html
https://devdocs.magento.com/guides/v2.3/config-guide/varnish/config-varnish-install.html
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/cli-cmds/cli-howto.html
https://devdocs.magento.com/guides/v2.4/test/integration/annotations.html  

## PR NOTES
Page Moved
- _site/cloud/cdn/fastly-vcl-badreferer.html

Content Restructured - Now added closest content
- _site/guides/v2.3/coding-standards/docblock-standard-general.html

Blog gone - found more recent article - https://www.keycdn.com/blog/x-frame-options
- _site/guides/v2.3/config-guide/secy/secy-xframe.html

4.1 is now removed from Varnish - have updated to 6.5 documentation links (and 5.2)
- _site/guides/v2.3/config-guide/varnish/config-varnish-advanced.html

Updated to Varnish's own WIKI
- _site/guides/v2.3/config-guide/varnish/config-varnish-install.html

Links Updated
- _site/guides/v2.3/extension-dev-guide/cli-cmds/cli-howto.html

Typo in URL
- _site/guides/v2.4/test/integration/annotations.html

Repo has moved to Github
- _site/mftf/v2/docs/test/annotations.html

### CODE REVIEW

File: src/guides/v2.3/config-guide/varnish/config-varnish-advanced.md [hunk 1]
```diff
@@ -82,8 +82,8 @@ bin/magento cache:flush
 
 Saint mode is not part of the main Varnish package. It is a separately-versioned vmod that must be downloaded and installed. As a result, you should re-compile Varnish from source, as described in the following articles:
 
-*  [Installing Varnish 5.1](https://varnish-cache.org/docs/5.1/installation/index.html)
-*  [Installing Varnish 4.1](https://varnish-cache.org/docs/4.1/installation/install.html) (Stable)
+*  [Installing Varnish 6.5](https://varnish-cache.org/docs/6.5/installation/index.html)
+*  [Installing Varnish 5.2](https://varnish-cache.org/docs/5.2/installation/index.html)
```
Comment (dshevtsov) [inline src/guides/v2.3/config-guide/varnish/config-varnish-advanced.md:86]:
```suggestion
```

5.x is not listed in [Technical requirements](https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements.html#technologies-magento-can-use)

**Review by dshevtsov — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/coding-standards/docblock-standard-general.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ The goal of this standard is to unify usage of code DocBlocks for all files, not
 
 The following is assumed by default:
 
-*  Formatting according to the [phpDocumentor](https://docs.phpdoc.org/latest/references/phpdoc/basic-syntax.html#what-is-a-docblock){:target="_blank"} standard
+*  Formatting according to the [phpDocumentor](https://docs.phpdoc.org/latest/guide/getting-started/your-first-set-of-documentation.html){:target="_blank"} standard
```
[the remark this answers is not in the document]
Comment (BarnyShergold) [reply src/guides/v2.3/coding-standards/docblock-standard-general.md:23]: Well found :)

File: src/guides/v2.3/config-guide/secy/secy-xframe.md [hunk 3]
```diff
@@ -57,4 +57,4 @@ Look for the `X-Frame-Options` value in the headers.
 
 *  [mozilla.org](https://developer.mozilla.org/en-US/docs/HTTP/X-Frame-Options)
 *  [security.stackexchange.com](http://security.stackexchange.com/questions/67889/why-do-browsers-enforce-the-same-origin-security-policy-on-iframes)
-*  [MSDN blog](http://blogs.msdn.com/b/ieinternals/archive/2010/03/30/combating-clickjacking-with-x-frame-options.aspx)
+*  [KeyCDN blog](https://www.keycdn.com/blog/x-frame-options)
```
[the remark this answers is not in the document]
Comment (BarnyShergold) [reply src/guides/v2.3/config-guide/secy/secy-xframe.md:60]: Sure, I can remove this - it was replacing an Blog link :) I'll remember for the future

File: src/guides/v2.3/config-guide/varnish/config-varnish-advanced.md (hunk 1 above)
Comment (BarnyShergold) [reply src/guides/v2.3/config-guide/varnish/config-varnish-advanced.md:86]: As Homer would say "Doh!"

**Review by dshevtsov — APPROVED** [review]
Great job!

**Comment by meker12** [comment]
running tests

**Pull request merged**